File: 4/foursum/src/pythonSol/simple.py
from __future__ import print_function
import sys

N = int(sys.stdin.readline())
vals = list(map(int, sys.stdin.readlines()))

for i in range(0, N):
    # your code goes here and uses the following
    for j in range(0, N):
        for k in range(0, N):
            for l in range(0, N):
                if vals[i]+vals[j]+vals[k]+vals[l] == 0:
                    print(i,j,k,l,file=sys.stderr)
                    print(True)
                    sys.exit()
print(False)

File: 4/foursum/src/javaSol/Simple.java
import java.util.Scanner;

public class Simple
{
    public static void main(String[] args)
    {
	Scanner S= new Scanner(System.in);
	int N = Integer.parseInt(S.nextLine());
	long[] vals = new long[N];
	for(int i= 0; i < N; i+= 1) {
		vals[i] = Long.parseLong(S.nextLine());
	}

        // your code goes here and uses the following
		for(int i= 0; i< N; i+= 1) {
			for(int j= i+1; j< N; j+= 1) {
				for(int k= j+1; k< N; k+= 1) {
					for(int l= k+1; l< N; l+= 1) {
						if (vals[i] + vals[j] + vals[k] + vals[l] == 0) 
							{
								System.err.println(i+" "+j+" "+k+" "+l);
								System.out.println(true);
								System.exit(0);
							}
					}
				}
			}
		}
			
        System.out.println(false);
    }
}